### PR TITLE
UI/UX: Display delivered message on incoming ACK.

### PR DESCRIPTION
**Pull request by caveman99**
Needs more work, see comments.

### CODE REVIEW

File: src/modules/CannedMessageModule.h [hunk 1]
```diff
@@ -37,15 +38,29 @@ class CannedMessageModule : public SinglePortModule, public Observable<const UIF
     const char *getMessageByIndex(int index);
     const char *getNodeName(NodeNum node);
     bool shouldDraw();
-    void eventUp();
-    void eventDown();
-    void eventSelect();
+    // void eventUp();
+    // void eventDown();
+    // void eventSelect();
```
Comment (caveman99) [inline src/modules/CannedMessageModule.h:44]: This is some leftover, testing if they are still needed

File: src/modules/CannedMessageModule.cpp [hunk 2]
```diff
@@ -546,6 +552,24 @@ void CannedMessageModule::drawFrame(OLEDDisplay *display, OLEDDisplayUiState *st
...
 
+ProcessMessage CannedMessageModule::handleReceived(const meshtastic_MeshPacket &mp)
+{
+    if (mp.decoded.portnum == meshtastic_PortNum_ROUTING_APP) {
+        // look for a request_id
+        if (mp.decoded.request_id != 0) {
+            UIFrameEvent e = {false, true};
+            e.frameChanged = true;
+            this->runState = CANNED_MESSAGE_RUN_STATE_ACK_RECEIVED;
+            this->incoming = mp.decoded.request_id;
```
Comment (caveman99) [inline src/modules/CannedMessageModule.cpp:563]: This is not actually the node number but the message id. We need to dig into the outgoing message store if we find the 'dest' field in it. Right now it always displays 'unknown'

Comment (GUVWAF) [reply src/modules/CannedMessageModule.cpp:568]: About this: we could for now just display "Last message delivered" or "Last delivery failed" until we have a way of finding the `dest` field?

File: src/mesh/MemoryPool.h
Comment (jp-bennett) [inline src/mesh/MemoryPool.h:1]: Is this intended to be part of this PR?

Comment (caveman99) [reply src/mesh/MemoryPool.h:1]: Yes this is intended. While researching alternative ways to find the originating message, I found out this is dead code. We can put it into a separate pr but I don't think it warrants the overhead.

**Review by GUVWAF — APPROVED** [review]
Looks good to me. 
I think we can eventually get rid of this special FreeRTOS queue for the `ToPhoneQueue` (seems this was actually [the plan](https://github.com/meshtastic/firmware/blob/master/src/concurrency/OSThread.h#L29)) to handle persistent storage for message history on standalone devices. If we can replace it by a `MeshPacketQueue`, which is actually a `std::vector`, we can more easily cycle through it and reorder packets.

**Comment by caveman99** [comment]
i pushed the current state to master, let's keep this branch open to refine this further. I'll push canned messaging to secondary channels next.

**Pull request merged**